source/models/row.py

```diff
@@ -14,32 +14,29 @@ def get_primary_key(self):
 
     def serialize(self):
         self.__lock()
-        self.__colomns_name_validate()
+        self.colomns_name_validate(self.get_value())
         row_json_data = json.dumps(self.__value, indent = 2)
-        with open(self.get_path(), 'w') as row_file: 
+        with open(self.get_path(self.get_primary_key()), 'w') as row_file: 
             row_file.write(row_json_data)
         for index_name in self.table.table_metadata.index_keys:
- 
             index = self.table.table_metadata.indicies[index_name]
             index.add_primary_key(primary_key = self.get_primary_key(), index_value = self.__value[index_name])
         self.__unlock()
 
     def row_exists(self):
-        return os.path.isfile(self.get_path())
+        return os.path.isfile(self.get_path(self.get_primary_key()))
 
-    def get_path(self):
-        return os.path.join(self.table.get_path(), os.path.join("Data", self.get_primary_key() + ".json"))
+    def get_path(self, primarykey):
```

Comment:
Why do you need to do this?

Reply:
read the comment below first . as you see joseph used "self.get_primary_key()" as it the primary key of the value and I don't have value so I edit it to take the the primarykey

Reply:
You should have two methods `get_path`

```
 public static String get_path(table, primary_key) {
    os.path(table.get_data_path(), primary_key)
}

public String get_path() {
   return Row.get_path(this.table, this.get_primary_key())
}
```

If you can't name the two method with the same name in python, name the static method `get_row_path`

Reply:
If you did this, u can revert all the changes in `get_path`

Reply:
i don't have value to get_primary_key() from it

Reply:
the same proplem i want to path the primary_key to it

Reply:
and don't use this.get_primary_key()